framework/src/interfaces/MortarInterface.C

```diff
@@ -55,6 +55,23 @@ MortarInterface::MortarInterface(const MooseObject * moose_object)
   if (_moi_mesh.dimension() == 3)
     mooseError("Mortar cannot currently be run in three dimensions. It's on the to-do list!");
 
+  if (_secondary_id == Moose::INVALID_BOUNDARY_ID)
+    mooseError(moose_object->getParam<BoundaryName>("secondary_boundary"),
+               " does not appear to be a boundary on the mesh. Did you make an error in your "
+               "mortar object input file block?");
+  if (_primary_id == Moose::INVALID_BOUNDARY_ID)
+    mooseError(moose_object->getParam<BoundaryName>("primary_boundary"),
+               " does not appear to be a boundary on the mesh. Did you make an error in your "
+               "mortar object input file block?");
+  if (_secondary_subdomain_id == Moose::INVALID_BLOCK_ID)
+    mooseError(moose_object->getParam<SubdomainName>("secondary_subdomain"),
+               " does not appear to be a subdomain on the mesh. Did you make an error in your "
+               "mortar object input file block?");
+  if (_primary_subdomain_id == Moose::INVALID_BLOCK_ID)
+    mooseError(moose_object->getParam<SubdomainName>("primary_subdomain"),
+               " does not appear to be a subdomain on the mesh. Did you make an error in your "
+               "mortar object input file block?");
+
```

Comment:
I'd make those assertions here and do the checking in the actual objects, where you can throw a param error, possibly with a wrapped getParam that does the checking